main.go

```diff
@@ -482,6 +498,9 @@ func Process(s *State) {
 	processorGroup.Wait()
 	close(resultChan)
 	printerGroup.Wait()
+	if s.OutputFileName != "" {
```

Comment:
Might also be worth checking `s.OutputFile` for `nil` rather than relying on the name.